src/graph.h

```diff
@@ -0,0 +1,60 @@
+/*	A Graph is an ordered set of vertices and edges.
+	Both, the Adjacency List and Adjacency Matrix implementation of
+	a graph are supported.
+
+	Usage :
+		Define a graph with 'struct Graph *'
+
+	Operations :
+		`INIT_GRAPH()`: Initiate an empty graph
+		`ADD_EDGE()`: Adds an edge between two nodes with a weight
+		`TO_LIST()`: List implementation of the graph
+		`TO_MATRIX()`: Matrix implementation of the graph
+
+	Example :
+*/
+
+#ifndef GRAPH_H
+#define GRAPH_H
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+
+#define LIST 1
+#define MATRIX -1
+#define BOTH 0
```

Comment:
This hackery can be replaced with the help of enumeration.
See this [enumeration example](http://stackoverflow.com/questions/1102542/how-to-define-an-enumerated-type-enum-in-c).

Basically, in place of these three lines, there should be
```c
typedef enum {LIST, MATRIX, BOTH} flag_type
```
and instead of `int flag` at L40, use
```
flag_type flag
```

This should work.